crates/sncast/src/lib.rs

```diff
@@ -278,22 +279,36 @@ pub async fn wait_for_tx(
     }
     let retries = timeout / u16::from(retry_interval);
     for i in (1..retries).rev() {
-        match provider.get_transaction_receipt(tx_hash).await {
-            Ok(receipt) => {
-                return match receipt.execution_result() {
-                    ExecutionResult::Succeeded => Ok("Transaction accepted"),
-                    ExecutionResult::Reverted { reason } => {
-                        Err(anyhow!("Transaction has been reverted = {reason}"))
+        match provider.get_transaction_status(tx_hash).await {
+            Ok(status) => match status {
+                starknet::core::types::TransactionStatus::Received => {
+                    println!("Transaction has been received");
```

Comment:
If there is a delay between the receipt and acceptance of a transaction I guess we may end up printing this message multiple times, which I think might not be an ideal outcome. What are your thoughts?

Reply:
hmm, that's true actually. Maybe we should combine Received with HashNotFound, as they are very similar? I think i raised this somewhere in this PR. That would make for simpler messaging probably.

Reply:
We can also include some logic to print the received message just once. As for `HashNotFound`, in my view we probably don't need to print a message at all. We could treat it as an expected error, which is not relevant for a user and simply retry after it.

Reply:
yup, that's what I had in mind. Basically we could just combine both matches into one, and only print the wait/timeout message

Reply:
relevant discussion https://github.com/foundry-rs/starknet-foundry/pull/1416#discussion_r1440442633

Reply:
There are nested matches, any idea how to do it better than I did?

Reply:
Rough idea would be to simply get rid of nested matches (for example match status can be removed). 

To be more precise:
- remove Ok(status) => match status altogether
- pack TransactionStatuses in Result (Ok())
- combine Received with TxHashNotFound
- profit

Let me know if everything is clear, and if you need any further help :+1: